crates/emitter/src/scope_pass.rs

```diff
@@ -411,7 +430,8 @@ impl GlobalContext {
         // Step 18. For each String vn in declaredVarNames, in list order, do
         for n in &self.declared_var_names {
             // 18.a. Perform ? envRec.CreateGlobalVarBinding(vn, false).
-            data.bindings.push(BindingName::new(*n))
+            let is_closed_over = closed_over_defs.contains(n);
```

Comment:
This code implies that we create a HashSet with all `closed_over_defs`, and query each name. I wonder whether it would be more efficient to not allocate memory for the temporary HashSet, and only query the `name_tracker`. Thus replacing `closed_over_defs.contains(n)` by `self.is_closed_over_def(n)`.

I presume that if we were to guarantee that both the `closed_over_names` and `defs_names` HashSet have the same hash function, we could query both sets in an efficiently without this temporary allocation.

Open a follow-up issue to address this question in terms of potential future optimization.